### PR TITLE
[FIX] l10n_at: tax_negate does now what this maybe was in the past

### PR DESCRIPTION
**Pull request by wtaferner**
**Description of the issue/feature this PR addresses:**
Correct the tax tags on the repartition line to have a proper tax report

**Current behavior before PR:**
Negative Base Amount presented in the tax report

**Desired behavior after PR is merged:**
Positive Base Amount presented in the tax report

Info: @wt-io-it

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by svfu-odoo — COMMENTED** [review]
Hello, thanks for the PR.
It makes sense to me and is consistent with the rest of the "VAT Computation" section of the tax report.

#### Potential problem after chart reload
But could this not cause problem on upgrade / for people who reload the chart template?
Example:
Before reload: someone has already `-1000 €` base amount in the tax report due to a `1000 €` purchase.
Then they reload the chart template (and the tags will be updated).
Then they make another purchase for `1000 €` but now there will be a positive sign on the base amount.
Then the tax report would say `0 €` for the base amount.

I guess mass-editing the lines would be a possibility.
But maybe this issue could be prevented in the first place?
Or how did you handle it or plan to handle it?

Or maybe it would be an option to just flip the sign on the tax report?

I will also check with the team next week.

#### Some other questions / suggestions

1. Should we not do this also in 16.0? The seem to show up there as a negative amount in the tax report as well. (I have not checked all taxes though)
   But I guess we can backport later too (since there the taxes are still in XML the PR is completely different anyway)
2. Regarding the commit title: What is the relevance of tax_negate?
   Please update the commit title to be a bit more direct description; maybe sth like: `correct the sign of some tax tags`.
3. I searched for similar cases and the following taxes seem to have the same issue:

   - `account_tax_template_purchase_eu_xx_code076`
   - `account_tax_template_purchase_eu_xx_vst_076`

   Or is there a reason they are different?

File: addons/l10n_at/data/template/account.tax-at.csv [hunk 1]
```diff
@@ -115,32 +115,32 @@
 "","","","","","","","","","","","","tax","invoice","","chart_at_template_3505","",""
 "","","","","","","","","","","","","base","refund","","","",""
 "","","","","","","","","","","","","tax","refund","","chart_at_template_3505","",""
-"account_tax_template_purchase_eu_0_code071","0% Ust","UST_071 IGE 0% (Art. 6 para. 2)","0%","200","purchase","","0.0","percent","False","","tax_group_0","base","invoice","-KZ 070||-KZ 071","","","UST_071 IGE 0% (Art. 6 Abs. 2)"
+"account_tax_template_purchase_eu_0_code071","0% Ust","UST_071 IGE 0% (Art. 6 para. 2)","0%","200","purchase","","0.0","percent","False","","tax_group_0","base","invoice",+KZ 070||+KZ 071,"","","UST_071 IGE 0% (Art. 6 Abs. 2)"
```
[the remark this answers is not in the document]
Comment (wtaferner) [reply addons/l10n_at/data/template/account.tax-at.csv:118]: Good find, I will do...

**Comment by svfu-odoo** [comment]
I checked / discussed it with the team.

And it seems that it was wrong in 15.0 and 16.0 as well.
I also "stepped" through the history of the tax csv / xml (in earlier version).
It seems like the tax tag on the base line was always negative in 17 and 16.
Are you sure there was a time were the tax tag was positive?

Currently we think it is not the case; so the sign was always the same.
So for us it is just the "sign" in the tax report that is wrong.
Which is not that big of a deal as you said:
Since we do not have an XML export the user just has to invert the sign
(nd a negative number will be rejected anyway).

In case we would introduce your change we would have the aforementioned problem
with mixed signs on the tags; which is much more messy.
I can see that fixing it is manageable for your customers and there is nothing inherently wrong with your fix.
But for other people the change may introduce unexpected issues (which should not happen in stable).
I.e. we do not want to mess with the users accounting entries automatically (like automatically running sth. like your script)
or possible changes they have done to fix the issue themselves.

cc: @tsb-odoo , @oco-odoo

**Comment by wtaferner** [comment]
@svfu-odoo 
Great, not happy about the refactoring and now applying stable policy, it was definitely not the same as I migrated from 15.0 to 17.0 and behavior changed for sure, but yeah, fine for me, here the cherry-picked and directed to master...no idea why there is still a "Conflicting files" dialog, but maybe it will disappear later...whatever, you can take over from here as it is 2,5 or 3 years away from my reality.

**Comment by svfu-odoo** [comment]
image

**Comment by wtaferner** [comment]
Ok, unklar wieso es da wirklich einen Unterschied gibt, aber gut, ich glaub die Lösung ist halt dann in 19.0 zu finden bzw. derzeit individuell zu beherrschen...so dringend und wichtig ist es IMO nicht, weil "nur" die Basis und die Meldung kann sowieso nur positiv und aktuell manuell erfolgen. 

Danke Dir und bitte asap in den Master mergen, damit wir das Thema abgehakt haben 😉

**Comment by oco-odoo** [comment]
The task has been created in the pipe for master. I'm closing this PR for now (because we need to create a new one in order to link it with the migration script properly), but we'll just cherry-pick your commit on our Community branch. Thanks for contributing :)

**Pull request closed without merging**